app/views/contexts/preview/preview_question.py

```diff
@@ -0,0 +1,117 @@
+from re import findall
+
+from app.questionnaire import QuestionnaireSchema
+
+
+class PreviewQuestion:
+    def __init__(self, question_schema, survey_data):
+        self.title = question_schema.get("title")
+        self.answers = self._build_answers(question_schema)
+        self.descriptions = self._build_descriptions(question_schema, survey_data)
+        self.guidance = self._build_question_guidance(question_schema)
+        self.text_length = self._get_length(question_schema.get("answers"))
+        self.instruction = question_schema.get("instruction") or None
+        self.answer_description = self._build_answer_descriptions(
+            iter(question_schema["answers"])
+        )
+        self.answer_guidance = self._build_answer_guidance(
+            answers=iter(question_schema["answers"])
+        )
+        self.survey_data = survey_data
+
+    @staticmethod
+    def _build_answers(question_schema):
+        answers = []
+        for answer in iter(question_schema["answers"]):
+            if options := answer.get("options"):
+                answers.extend(option["label"] for option in options)
```

Comment:
There is a similar issue here as I commented on Previously, in that it is possible do have a placeholder in an answer option, like when we pipe in an address, so this would need to be handled as well.

Reply:
Options placeholders are now being resolved using the same method we use for resolving question descriptions, thanks.